src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,43 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public LocalDate getLocalDate(String date) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        return LocalDate.parse(date, formatter);
+    }
+
+    public boolean isValidDate(LocalDate startDate, LocalDate endDate, String currentDate) {
+        LocalDate localDate = new SalaryInfo().getLocalDate(currentDate);
```

Comment:
there is no point in creating a new instance of SalaryInfo when you are in one of its methods, you can just call the method directly. There is already an instance of SalaryInfo created by someone, which will call the method getSalaryInfo() and expect some results. Your task here is to implement the logic which will provide this result.